apollo-federation/src/merge.rs

```diff
@@ -222,6 +230,7 @@ impl Merger {
                     ExtendedType::Object(value) => self.merge_object_type(
                         &mut supergraph.types,
                         &relevant_directives,
+                        &composed_directives_names,
```

Comment:
This PR is currently only passing the names of directives specified in @composeDirective to `merge_object_type` . This needs to be passed to other merge fn's as well of course.